web/src/lib/components/photos-page/actions/delete-assets.svelte

```diff
@@ -7,39 +7,45 @@
 	import { api } from '@api';
 	import DeleteOutline from 'svelte-material-icons/DeleteOutline.svelte';
 	import { OnAssetDelete, getAssetControlContext } from '../asset-select-control-bar.svelte';
+	import ConfirmDialogue from '$lib/components/shared-components/confirm-dialogue.svelte';
+	import { handleError } from '../../../utils/handle-error';
 
 	export let onAssetDelete: OnAssetDelete;
 	const { getAssets, clearSelect } = getAssetControlContext();
 
-	const deleteSelectedAssetHandler = async () => {
+	let confirm = false;
+
+	const handleDelete = async () => {
 		try {
-			if (
-				window.confirm(
-					`Caution! Are you sure you want to delete ${
-						getAssets().size
-					} assets? This step also deletes assets in the album(s) to which they belong. You can not undo this action!`
-				)
-			) {
-				const { data: deletedAssets } = await api.assetApi.deleteAsset({
-					ids: Array.from(getAssets()).map((a) => a.id)
-				});
-
-				for (const asset of deletedAssets) {
-					if (asset.status === 'SUCCESS') {
-						onAssetDelete(asset.id);
-					}
-				}
+			let count = 0;
 
-				clearSelect();
+			const { data: deletedAssets } = await api.assetApi.deleteAsset({
+				ids: Array.from(getAssets()).map((a) => a.id)
+			});
+
+			for (const asset of deletedAssets) {
+				if (asset.status === 'SUCCESS') {
+					onAssetDelete(asset.id);
+					count++;
+				}
 			}
+
+			notificationController.show({ message: `Deleted ${count}`, type: NotificationType.Info });
+
+			clearSelect();
 		} catch (e) {
-			notificationController.show({
-				type: NotificationType.Error,
-				message: 'Error deleting assets, check console for more details'
-			});
-			console.error('Error deleteSelectedAssetHandler', e);
+			handleError(e, 'Error deleting assets');
 		}
 	};
 </script>
 
-<CircleIconButton title="Delete" logo={DeleteOutline} on:click={deleteSelectedAssetHandler} />
+<CircleIconButton title="Delete" logo={DeleteOutline} on:click={() => (confirm = true)} />
+
+{#if confirm}
+	<ConfirmDialogue
+		prompt="Are you sure you want to delete {getAssets()
+			.size} assets? This step also deletes assets in the album(s) to which they belong. You can not undo this action!"
+		on:confirm={handleDelete}
```

Comment:
For the confirm button it's generally recommended to choose a more specific label like delete or update over generic ones like save, yes and confirm

```suggestion
		title="Delete assets?"
		confirmText="Delete"
		on:confirm={handleDelete}
```

Reply:
This is used several places. It may be worth revisiting then as well.